src/gui/folderman.cpp

```diff
@@ -720,22 +720,41 @@ static QString canonicalPath(const QString &path)
     return selFile.canonicalFilePath();
 }
 
-static QString checkPathForSyncRootMarkingRecursive(const QString &path)
-{
-    auto existingTag = Utility::getDirectorySyncRootMarking(path);
-    if (!existingTag.isEmpty()) {
-        return FolderMan::tr("Folder '%1' is already in use by application %2!").arg(path, existingTag);
+static QString checkPathForSyncRootMarkingRecursive(const QString &path, FolderMan::NewFolderType folderType, const QUuid &accountUuid)
+{
+    std::pair<QString, QUuid> existingTags = Utility::getDirectorySyncRootMarkings(path);
+    if (!existingTags.first.isEmpty()) {
+        if (folderType == FolderMan::NewFolderType::SpacesFolder) {
+            if (existingTags.first == Theme::instance()->orgDomainName()) {
+                // Looks good, our app, let's check the account:
+                if (existingTags.second == accountUuid) {
+                    // Nice, that's what we like, the sync root for our account in our app. No error.
+                    return {};
+                } else {
+                    return FolderMan::tr("Folder '%1' is already in use by another account.").arg(path);
+                }
+            } else {
+                // this is handled below: another application uses this as spces root folder
+            }
+        }
+
+        if (existingTags.first == Theme::instance()->orgDomainName()) {
+            // it's our application, so must be another space root
+            return FolderMan::tr("Folder '%1' is already in use by another account.").arg(path);
+        }
+        // another application uses this as spaces root folder
+        return FolderMan::tr("Folder '%1' is already in use by application %2!").arg(path, existingTags.first);
```

Comment:
I think it could make things easier to parse if we handle this scenario first? It would make the two checks for domain superfluous.

Reply:
done